src/ucp/proto/proto_perf.c

```diff
@@ -430,6 +430,28 @@ ucs_status_t ucp_proto_perf_aggregate2(const char *name,
     return ucp_proto_perf_aggregate(name, perf_elems, 2, perf_p);
 }
 
+void ucp_proto_perf_apply_bias(ucp_proto_perf_t *perf, double bias)
+{
+    ucs_linear_func_t bias_func = ucs_linear_func_make(0.0, 1.0 - bias);
+    ucp_proto_perf_node_t *bias_node;
+    ucp_proto_perf_factor_id_t fid;
+    ucp_proto_perf_segment_t *seg;
+
+    if (fabs(bias) <= UCP_PROTO_PERF_EPSILON) {
+        return;
+    }
+
+    ucp_proto_perf_segment_foreach(seg, perf) {
+        for (fid = 0; fid < UCP_PROTO_PERF_FACTOR_LAST; ++fid) {
+            seg->perf_factors[fid] =
```

Comment:
maybe better use a local array for the new factors value rather than update in place since ucp_proto_perf_node_update_factors does not expect them in place (though it works)